### DISCUSSION TITLE
Close TabTabTab when switching to another window

### DISCUSSION THREAD

**Discussion by FreePDF64** [okaryo/TabTabTab#249] [Ideas]
Hello and Merry Christmas :)

I have noticed the following: 
I have two windows with tab groups in them. Let's call it window1 and window2. When I switch from window1 via TabTabTab to window2, the popup in window1 stays open and still shows me the number of tabs in window2. If I now switch to window2 and close some tabs there, window1 with TabTabTab still open does not notice this and continues to show me the old number of tabs of window2. If I now switch back to window1, I cannot access the tabs of window2 that no longer exist via TabTabTab - of course.

Perhaps a workaround would be for TabTabTab to close itself when switching windows?

I hope you understand me, because it is difficult to create a video on this topic...

Best regards,
Michael

**Comment by okaryo** [comment]
Hi Michael,

Merry Christmas to you too!

I've noticed the issue you described with the TabTabTab extension across different windows. You're right, it's not the best experience when the popup in one window doesn't update after changes are made in another.

After a quick review, I believe that automatically closing the popup when switching windows could be a good solution. I'll work on a fix and update the extension soon.

Thank you for bringing this to my attention. It's really helpful to receive feedback on even the small details. It helps us improve.

Best regards.

**Reply by okaryo** [reply]
done by #250
